control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -3415,6 +3416,33 @@ func (r *HostedControlPlaneReconciler) reconcileOAuthServer(ctx context.Context,
 	}); err != nil {
 		return fmt.Errorf("failed to reconcile oauth deployment: %w", err)
 	}
+
+	// Report any IDP configuration errors as a condition on the HCP
+	orig := meta.FindStatusCondition(hcp.Status.Conditions, string(hyperv1.ValidIDPConfiguration))
+	new := &metav1.Condition{
+		Type:    string(hyperv1.ValidIDPConfiguration),
+		Status:  metav1.ConditionTrue,
+		Reason:  "IDPConfigurationValid",
+		Message: "Identity provider configuration is valid",
+	}
+	if _, _, err := oauth.ConvertIdentityProviders(ctx, p.IdentityProviders(), p.OauthConfigOverrides, r, hcp.Namespace); err != nil {
+		// Report the error in a condition on the HCP
+		new = &metav1.Condition{
+			Type:    string(hyperv1.ValidIDPConfiguration),
+			Status:  metav1.ConditionFalse,
+			Reason:  "IDPConfigurationError",
+			Message: fmt.Sprintf("failed to initialize identity providers: %v", err),
+		}
+	}
+	// Update the condition on the HCP if it has changed
+	if !reflect.DeepEqual(orig, new) {
```

Comment:
This would always be false because orig will have a LastTransitionTime, no?

Reply:
Also, could we get in a hot loop if the error returned is slightly different every time?

Reply:
> This would always be false because orig will have a LastTransitionTime, no?

yes, I do need to clear that before the comparison

> Also, could we get in a hot loop if the error returned is slightly different every time?

I thought about this.  We could just update when the Status changes.  However, the user moves from one invalid config to another and the error changes, we wouldn't be reporting the new error.  :thinking:

Reply:
`meta.SetStatusCondition()` retrieves the existing condition, compares it with the new condition and returns a bool if the condition changed, you shouldn't need to retrieve orig condition to compare manaully.
```golang
changed := meta.SetStatusCondition(&hcp.Status.Conditions, new)
if changed {
    if err := r.Status().Update(ctx, hcp); err != nil {
	return fmt.Errorf("failed to update valid IDP configuration condition: %w", err)
    }
}
```